src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.SingleEntry.targets

```diff
@@ -2,24 +2,19 @@
 
   <PropertyGroup Condition="'$(RuntimeIdentifier)' != '' and '$(NETCoreSdkPortableRuntimeIdentifier)' != ''">
     <!-- Define the name of the runtime specific compiler package to import -->
-    <_targetOSPkg>$(NETCoreSdkPortableRuntimeIdentifier.SubString(0, $(NETCoreSdkPortableRuntimeIdentifier.LastIndexOf('-'))))</_targetOSPkg>
-    <_indexOfHyphen>$(_targetOSPkg.IndexOf('-'))</_indexOfHyphen>
-    <_targetOS Condition="'$(_indexOfHyphen)' &gt; -1">$(_targetOSPkg.SubString(0, $(_indexOfHyphen)))</_targetOS>
-    <_targetOS Condition="'$(_indexOfHyphen)' == -1">$(_targetOSPkg)</_targetOS>
+    <_targetOS>$(NETCoreSdkPortableRuntimeIdentifier.SubString(0, $(NETCoreSdkPortableRuntimeIdentifier.IndexOf('-'))))</_targetOS>
```

Comment:
`NETCoreSdkPortableRuntimeIdentifier` is the host RID. `RuntimeIdentifier` is the target RID.

Should we rather use `RuntimeIdentifier` to compute the `_targetOS` here?

Reply:
(I know we have switch to `NETCoreSdkPortableRuntimeIdentifier` for the portable Unix, but it looks like that it just replaced one problem with a different one.)

Reply:
Per docs https://github.com/dotnet/runtime/blob/bd2e520e5960f5d6f0bf27fd51ec5d2128f49989/src/coreclr/nativeaot/docs/compiling.md#cross-architecture-compilation:

> Cross-OS compilation, such as targeting Linux on a Windows host, is not supported.

So up until very recently, we weren't supporting cross-OS NativeAOT publish (only cross-Architecture of same OS), so it works for those cases and makes targetOS distro-agnostic.

To support new scenarios like "publish for iOS from OSX", it requires making this infra aware of cross-OS publishing. We still don't support cross-OS compilation for anything other than OSX -> other Apple OS, so we can special case that here. I'm thinking about how to handle it in a scalable way.

Reply:
> scalable way

With `TargetPlatformIdentifier` conditions in place, it handles both "distro-agnostic RIDs" and "Xamarin SDK platforms".